flash/data/data_module.py

```diff
@@ -53,20 +56,35 @@ class DataModule(pl.LightningDataModule):
 
     def __init__(
         self,
-        train_dataset: Optional[Dataset] = None,
-        val_dataset: Optional[Dataset] = None,
-        test_dataset: Optional[Dataset] = None,
-        predict_dataset: Optional[Dataset] = None,
+        data_source: DataSource,
+        preprocess: Optional[Preprocess] = None,
+        postprocess: Optional[Postprocess] = None,
+        data_fetcher: Optional[BaseDataFetcher] = None,
+        val_split: Optional[float] = None,
         batch_size: int = 1,
         num_workers: Optional[int] = 0,
     ) -> None:
 
         super().__init__()
+
+        self._preprocess: Optional[Preprocess] = preprocess
+        self._postprocess: Optional[Postprocess] = postprocess
+        self._viz: Optional[BaseVisualization] = None
+        self._data_fetcher: Optional[BaseDataFetcher] = data_fetcher or self.configure_data_fetcher()
+
+        # TODO: Preprocess can change
+        self.data_fetcher.attach_to_preprocess(self.preprocess)
+
+        train_dataset, val_dataset, test_dataset, predict_dataset = data_source.to_datasets(self.data_pipeline)
```

Comment:
This line is confusing me. Why do we provide the data_pipeline ?

Reply:
Done :)